pandas/core/frame.py

```diff
@@ -3518,10 +3539,13 @@ def query(self, expr: str, inplace: bool = False, **kwargs):
 
         if inplace:
             self._update_inplace(result)
+            return None
         else:
             return result
 
-    def eval(self, expr: str, inplace: bool = False, **kwargs):
+    def eval(
+        self, expr: str, inplace: bool = False, **kwargs
+    ) -> Optional[Union[AnyArrayLike, DataFrame, Scalar]]:
```

Comment:
same remove this for now leave for another pass.

Reply:
Removed